swarm/network/stream/stream.go

```diff
@@ -992,3 +1037,26 @@ func (api *API) GetPeerSubscriptions() map[string][]string {
 	}
 	return pstreams
 }
+
+func (api *API) GetPeerClientSubscriptions() map[string][]string {
```

Comment:
Maybe to write a test for this function as well?

Reply:
This function is trivial, just like the `GetPeerServerSubscriptions`. I don't think we need tests for it, chances that someone will refactor it and break it are so slim, that I'd rather we don't add 50 lines of tests like the other one.

Also this is not an API call that is part of critical functionality, it is here for debug purposes.

If we add unit test for this, we will probably have to remove it, once we review the `Stream` implementation - something that has been in discussions over the last weeks.

Reply:
Actually it is not 50 lines, but much more, we also have `TestGetServerSubscriptionsRPC`

Reply:
`TestGetServerSubscriptionsRPC` depends on:
- `simulations` package
- `adapters`
- `netStore`
- `delivery`
- spec for `SubscribeMsg`
- json snapshots
- exact naming of the function, not covered by the compiler by dynamically linked

I'd rather we don't write such low-value tests for such trivial functionalities.

Reply:
Well, tools that can be used for testing the code should not be relevant for decision if the code should be tested. If this code exists and somebody refactors the stream package, I suppose that it would be nice to have a test to verify that there is no regression. I would not block this PR for not testing this code if it is not an issue for others.

Reply:
@janos I strongly disagree with this.

If you are using 10 libraries to test a `for` loop that is adding values to a slice, you are increasing the complexity of the codebase for no good reason.

Now that these tests are part of the codebase, every time we change any of those 10 libraries, we also have to modify the tests. This is a cost to development that we should not ignore. The benefit of this test is that we are sure that we have not broken the `for` loop that is adding values to a slice - rather low-value in my opinion.

Reply:
Thanks for explanations. I will not insist on this test.

Reply:
@janos giving a bit more thought to this particular piece of code - the sole purpose of this API call exists is to be a helper method to debug the already broken subscriptions implementation.

I'd rather we focus on solving the bugs in the subscriptions implementation (without commenting what this involves - rewrite or bug fix or whatever), than write tests for the helper methods we add to the codebase to help us track the bigger issues in Swarm.

This feels like adding a test for a metrics counter or timer to me.

So if we insist to have a test on this API (which I see you don't in the comment above), I'd rather just remove it altogether, and find another way to debug the subscriptions functionality.